app/domain/config-types.ts

```diff
@@ -119,14 +132,27 @@ export type BarConfig = t.TypeOf<typeof BarConfig>;
 
 const ColumnFields = t.intersection([
   t.type({
-    x: GenericField,
+    x: t.type({
+      componentIri: t.string,
+      sorting: t.type({
+        sortingField: t.union([t.literal("alphabetical"), t.literal("y")]),
+        sortingOrder: t.union([t.literal("asc"), t.literal("desc")]),
+      }),
+    }),
     y: GenericField,
   }),
   t.partial({
     segment: t.type({
-      componentIri: t.string,
+      componentIri: t.union([t.string, t.undefined]),
```

Comment:
Not sure if it's correct to let the `componentIri` be optional here. The whole `segment` field is already optional, and I don't think it makes much sense to have `segment` options without a componentIri.

Reply:
The reason I did that is to have a default sorting type and sorting order. In the stacked column chart, if selecting a dimension for the "partition", then "None", then a dimension again, the app was previously crashing because no default was set for the sorting options. I updated the action `CHART_OPTION_CHANGED` consequently.
I actually now updated this type back to `string`, and use in the app either a `dimensionIri` or `FIELD_VALUE_NONE`... I think we should actually go all the way and make segment mandatory (not a `partial`).

Reply:
Oh... but I see your previous point about not breaking previous chart configs...